### PR TITLE
Datastore demo

### PR DESCRIPTION
**Pull request by kingsleyzissou**
### Description

- Add late binding for datastore replication
- Add network hook
- Add volunteer entry model for offline write
- update some dependencies

### CODE REVIEW

**Comment by wtrocki** [comment]
Checking

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
Amazing work @kingsleyzissou

File: platform/client/src/context/AuthContext.tsx [hunk 1]
```diff
@@ -18,6 +19,13 @@ export const AuthContext = React.createContext<IAuthContext>({
 export const AuthContextProvider = (props: any) => {
     const setVolunteer = (volunteer: VolunteerType) => {
       setState({...state, volunteer: volunteer})
+      if (volunteer?._id) {
+        const filter = {
+          volunteerId: { eq: volunteer._id }
+        };
+        VolunteerActionModel.applyFilter(filter);
+        datastore.startReplication();
```
Comment (wtrocki) [inline platform/client/src/context/AuthContext.tsx:27]: Love it. This could land in the docs as well.
Really good job

**Review by wtrocki — APPROVED** [review]

File: platform/client/src/context/AuthContext.tsx (hunk 1 above)
Comment (kingsleyzissou) [reply platform/client/src/context/AuthContext.tsx:27]: Awesome, yeah I can add this to the docs :)

**Pull request merged**